client/shared/src/main/scala/org/http4s/client/digest/NonceCounter.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2014 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.client.digest
+
+import cats.effect.kernel.Async
+import cats.effect.std.Random
+import cats.syntax.all._
+
+import scala.concurrent.duration._
+
+trait NonceCounter[F[_]] {
+
+  /** Returns `cnonce` and next `nc` value for the given nonce */
+  def next(nonce: String): F[(String, Long)]
+
+  /** Call when a conversation is over, to prevent unbounded growth. */
+  def evict(nonce: String): F[Unit]
+
+  /** Call when a conversation is over, to prevent unbounded growth. */
+  def evictAll: F[Unit]
+}
+
+object NonceCounter {
+  val MaxNonce: Long = java.lang.Long.parseLong("FFFFFFFF", 16)
+
+  def make[F[_]: Async](
+      ttl: FiniteDuration = 1.minute,
+      maxNonce: Long = MaxNonce,
+  ): F[NonceCounter[F]] =
+    Random
+      .javaSecuritySecureRandom[F]
```

Comment:
According to the relative [RFC](https://datatracker.ietf.org/doc/html/rfc7616), there's no strict need for a cryptographically secure cnonce - however I think this won't hurt.

Reply:
Is this the relevant section?
https://datatracker.ietf.org/doc/html/rfc7616#section-5.12

> The security of this protocol is critically dependent on the
   randomness of the randomly chosen parameters, such as client and
   server nonces.  These should be generated by a strong random or
   properly seeded pseudorandom source (see [[RFC4086](https://datatracker.ietf.org/doc/html/rfc4086)]).

Reply:
yes, I think the `properly seeded pseudorandom` seems covered by the standard `Random` typeclass.

Reply:
Same! I'm only interpreting the RFC text, but I'm no expert and I might be blatantly wrong 😊